src/components/panel/panel.spec.js

```diff
@@ -542,6 +542,20 @@ describe('$mdPanel', function() {
       expect(PANEL_EL).not.toExist();
     });
 
+    it('should close when clickOutsideToClose set to true and ' +
+        'propagateContainerEvents is also set to true', function() {
+          var config = {
+            propagateContainerEvents: true,
+            clickOutsideToClose: true
+          };
+
+          openPanel(config);
+
+          clickOutsideOfPanel();
```

Comment:
Can this be clickPanelContainer? Or can clickPanelContainer be updated so you're not duplicating logic. (The only thing that's changing is the element you're clicking on. Maybe something like this?

```
function clickPanelContainer(rootEl) {
  rootEl = rootEl || document.body;
  const selector = '.md-panel:not(._md-panel-backdrop)';
  angular.element(rootEl).find(selector).parent().simulate('click');
  angular.element(rootEl).find(selector).parent().simulate('mousedown');
  angular.element(rootEl).find(selector).parent().simulate('mouseup');
  flushPanel();
}
```
Benefits of this version is that you don't need the panelRef, which removes one of the if statements. I know this version can replace the current one and you won't need to change any tests.

or 

```
function clickPanelContainer(container) {
    container = container || angular.element('.md-panel-outer-wrapper');

    container.triggerHandler({
      type: 'mousedown',
      target: container[0]
    });

    container.triggerHandler({
      type: 'mouseup',
      target: container[0]
    });

    flushPanel();
  }
```